core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -158,11 +157,6 @@
     private static int queryNameId = 0;
     private static SearchModule searchModule;
 
-
-    protected static QueryShardContext queryShardContext() {
-        return queryShardContext;
```

Comment:
it gets a bit weird when we use it to set stuff to the SearchContext although we create another context for those operations. That seems confusing... plus if it was not for templating, it would be only needed in init() which could have it a local variable. Maybe rename it at least so it is clearer why it's there? and fix some of its usages?